src/http_client_async.zig

```diff
@@ -3480,21 +3480,16 @@ pub fn handleResponseMetadata(
                     } else {
                         var url_buf = URLBufferPool.get(default_allocator);
                         const original_url = this.url;
-                        const port = original_url.getPortAuto();
-
-                        if (port == original_url.getDefaultPort()) {
-                            this.url = URL.parse(std.fmt.bufPrint(
-                                &url_buf.data,
-                                "{s}://{s}{s}",
-                                .{ original_url.displayProtocol(), original_url.displayHostname(), location },
-                            ) catch return error.RedirectURLTooLong);
-                        } else {
-                            this.url = URL.parse(std.fmt.bufPrint(
-                                &url_buf.data,
-                                "{s}://{s}:{d}{s}",
-                                .{ original_url.displayProtocol(), original_url.displayHostname(), port, location },
-                            ) catch return error.RedirectURLTooLong);
-                        }
+
+                        const new_url_ = bun.JSC.URL.join(
```

Comment:
Not sure using `JSC.URL.join`, maybe there's another function can be used instead.

Reply:
URL.join is probably the right move here. But it will need to handle the case where the URL is invalid